sql/src/main/java/io/crate/analyze/expressions/ExpressionAnalyzer.java

```diff
@@ -1054,6 +1067,7 @@ static Symbol allocateFunction(String functionName,
     private static Symbol allocateBuiltinOrUdfFunction(@Nullable String schema,
                                                        String functionName,
                                                        List<Symbol> arguments,
+                                                       @Nullable Symbol filter,
```

Comment:
it is growing a bit, is it ok to maybe clean it up later in `ma` or we should do it now?

Reply:
I think it is okay. We could look into reducing the amount of different functions / overloads. But isn't a big issue